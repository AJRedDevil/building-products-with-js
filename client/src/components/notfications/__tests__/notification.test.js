// npm packages
import React from 'react';
import configureMockStore from 'redux-mock-store';

// our packages
import NotificationWrapper, {Notification} from '../notification';

// create mockstore
const mockStore = configureMockStore();

const notification = {
  id: 0,
  text: 'Test',
  alertType: 'info',
};

test('# NotificationWrapper', () => {
  const store = mockStore({});
  const wrapperEmptyNotification = shallow(<NotificationWrapper store={store} />);
  expect(wrapperEmptyNotification).toMatchSnapshot();

  const wrapperWithNotification = shallow(<NotificationWrapper notification={notification} store={store} />);
  expect(wrapperWithNotification).toMatchSnapshot();
});

test('# Notification', () => {
  const onClick = id => expect(id).toBe(notification.id);
  const component = <Notification onRemoveNotificationClick={onClick} />;
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
  // test interaction
  const app = mount(component);
  const item = app.find('button');
  item.simulate('click');
});

