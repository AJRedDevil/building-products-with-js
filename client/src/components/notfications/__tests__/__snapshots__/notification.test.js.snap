// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Notification 1`] = `
<Fade
  in={true}
>
  <div
    className="alert alert-dismissible alert-"
    role="alert"
  >
    <button
      ari-label="Close"
      className="close"
      data-dismiss="alert"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
    </button>
  </div>
</Fade>
`;

exports[`# NotificationWrapper 1`] = `
<Notification
  notification={
    Object {
      "alertType": "",
      "id": 0,
      "text": "",
    }
  }
  onRemoveNotificationClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`# NotificationWrapper 2`] = `
<Notification
  notification={
    Object {
      "alertType": "info",
      "id": 0,
      "text": "Test",
    }
  }
  onRemoveNotificationClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
